### PR TITLE
feat: add codecov support for C++ bazel

### LINKED ISSUE #9296: Enable code coverage with Bazel

### PR DESCRIPTION
**Pull request by themarwhal**
Signed-off-by: Marie Bremner <marwhal@fb.com>

## Summary
1. (bazel config) Add java toolchain information as it is required for codecov
2. (CI config) Add bazel based codecov uploading job to upload targets that can be built with Bazel (we can migrate C/C++ codecov logic for sctpd/mme/liagent/ once they have been bazelified into this job)
3. (lte/gateway/Makefile) Modify existing CI job to upload C++ codecov to upload codecov for targets not bazelified

## Test Plan
Updated the codecov upload job on CI to use this for orc8r/gateway/c/common and lte/gateway/c/session_manager. We can slowly migrate to this job as we add more bazel targets for C/C++.

Screen Shot 2021-11-03 at 9 37 29 PM

## Benefit
With bazel coverage, we're able to capture test coverage for the orc8r/gateway/c/common directory. This is because we also run the tests in those directories to compute coverage as well. 

Codecov before: https://codecov.io/gh/magma/magma/tree/5fec3827dce5c9b9bfe49314790dbdc244e3ae7b
Screen Shot 2021-11-04 at 10 48 30 AM
Screen Shot 2021-11-04 at 10 48 43 AM

After
Screen Shot 2021-11-04 at 11 04 17 AM
Screen Shot 2021-11-04 at 1 09 37 PM

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -5,6 +5,18 @@ build --color=yes
 build --cxxopt=-std=c++14
 build --cxxopt=-g2
 
+build --javacopt="-source 8"
```
Comment (electronjoe) [inline .bazelrc:10]: I know this is a draft! But thought I would drop a question while Im' here.  Why the java stuff?

Comment (electronjoe) [reply .bazelrc:10]: Oh your PR description says why. Sad that it's required - how large an additional fetch is this?

Comment (themarwhal) [reply .bazelrc:10]: yes still trying to figure this out. But somehow the codecov doesn't work without java. It complains about not being able to find java :/ still trying to understand why though.

Comment (themarwhal) [reply .bazelrc:10]: good question.... can research and update the PR.

File: .github/workflows/agw-workflow.yml [hunk 2]
```diff
@@ -424,7 +426,11 @@ jobs:
         run: |
             docker build --tag magma/c_cpp_build --file ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme/Dockerfile.ubuntu20.04 .
             ci_env=$(bash <(curl -s https://codecov.io/env))
-            docker run $ci_env --env CI=true --volume ${{ env.MAGMA_ROOT }}:/magma --volume ${{ env.MAGMA_ROOT }}/lte/gateway/configs:/etc/magma --interactive magma/c_cpp_build:latest /bin/bash -c "cd /magma/lte/gateway;make coverage;ls -al /tmp/;bash <(curl -s https://codecov.io/bash) -f /build/c/coverage.info -F c_cpp"
+            docker run $ci_env --env CI=true --volume ${{ env.MAGMA_ROOT }}:/magma --volume ${{ env.MAGMA_ROOT }}/lte/gateway/configs:/etc/magma --interactive magma/c_cpp_build:latest /bin/bash -c "cd /magma/lte/gateway;make coverage;cp /build/c/coverage.info /magma/"
+      - name: Upload code coverage
+        uses: codecov/codecov-action@v1
+        with:
+          flags: c_cpp
```
Comment (electronjoe) [inline .github/workflows/agw-workflow.yml]: linter: trailing whitespace

**Review by electronjoe — APPROVED** [review]

**Comment by themarwhal** [comment]
@quentinDERORY and I chatted offline that we want to reduce the number of jobs on CI. Once https://github.com/magma/magma/pull/10114 lands, I can merge this codecov job into the unit testing job.

**Comment by themarwhal** [comment]
merging this change into https://github.com/magma/magma/pull/10114

**Pull request closed without merging**